src/System.Linq.Queryable/src/System/Linq/Queryable.cs

```diff
@@ -842,6 +842,23 @@ public static TSource SingleOrDefault<TSource>(this IQueryable<TSource> source,
                     ));
         }
 
+        public static bool TrySingle<TSource>(this IQueryable<TSource> source, out TSource element)
+        {
+            Tuple<bool, TSource> result = TrySingle(source);
+            element = result.Item2;
+            return result.Item1;
+        }
+
+        internal static Tuple<bool, TSource> TrySingle<TSource>(this IQueryable<TSource> source)
+        {
+            if (source == null)
+                throw Error.ArgumentNull(nameof(source));
+            return source.Provider.Execute<Tuple<bool, TSource>>(
+                Expression.Call(
+                    null,
+                    CachedReflectionInfo.TrySingle_TSource_1(typeof(TSource)), source.Expression));
+        }
+
```

Comment:
@bartdesmet If you did not find solution than I can suggest next. Make methods with Tuple in Enumerable public but not visible to editor. See EditorBrowsable(EditorBrowsableState.Never) attribute